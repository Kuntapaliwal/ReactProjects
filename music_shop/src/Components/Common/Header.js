import React, { Component } from 'react'
import{Link} from 'react-router-dom'
import '../../Styles/Header.css'
export default class Header extends Component {
  render() {
    return (
      <div className='header'>Music Shop
      <div >
        <Link to="/">Home</Link> | {" "}
        <Link to="/music">Music</Link>  | {" "}
        <Link to="/about">About</Link>  | {" "}
      </div>
      </div>
    )
  }
}
