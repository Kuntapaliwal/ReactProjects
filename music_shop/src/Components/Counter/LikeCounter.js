import React, { useState,useEffect } from 'react'

export default function () {
const[likes,setLikes]=useState(0)//initial value as argument likes is one variable and 2nd setlikes is method 
useEffect(()=>{
console.log("component Did mount")
},[])//behave like componenetDidMount if [] is blank


useEffect(()=>{
console.log("Compone is update")
},[likes])


useEffect(()=>{
  console.log("i will be called everytime")
})


useEffect(()=>{
  return ()=>console.log("component is unmounting")
})
  return (
   
    <div >
    <button className='btn btn-warning' onClick={() => setLikes(likes + 1)}>Like</button>
     <button className='btn btn-warning' onClick={() => setLikes(likes - 1)}>Unlike</button>
    {likes}
    </div>
    
  )
}
