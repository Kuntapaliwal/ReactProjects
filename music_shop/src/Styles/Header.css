.header{
    text-align: center;
    color: red;
    font-family: cursive;
    font-weight: bold;
    font-size: 2em;
}