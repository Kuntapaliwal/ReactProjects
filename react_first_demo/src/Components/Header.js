import '../Styles/Header.css'

function Header(props)//this props name is used for better understsanding we can take it any other name also
{
    return (
        <div className="header">
        <h2>I Am the header of your page {props.title}</h2>
        <p>{props.t2}</p>
        <p>{props.t3}</p>
        </div>
    )
    
    
}
export default Header;