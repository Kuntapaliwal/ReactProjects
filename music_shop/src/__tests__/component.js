import render from 'react-test-renderer'
import LikeCounter from '../Components/Counter/LikeCounter'
import Header from '../Components/Common/Header'
import {BrowserRouter} from 'react-router-dom'
describe('component testing',()=>{
it('test Like component',()=>{
    const tree=render.create(<LikeCounter/>).toJSON()
    expect(tree).toMatchSnapshot();
})

it('test component having link',()=>{
const tree=render.create(<BrowserRouter><Header/></BrowserRouter>).toJSON()
expect(tree).toMatchSnapshot();
})

})