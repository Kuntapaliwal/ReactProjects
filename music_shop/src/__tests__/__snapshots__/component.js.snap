// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component testing test Like component 1`] = `
<div>
  <button
    className="btn btn-warning"
    onClick={[Function]}
  >
    Like 56575
  </button>
  <button
    className="btn btn-warning"
    onClick={[Function]}
  >
    Unlike
  </button>
  0
</div>
`;

exports[`component testing test component having link 1`] = `
<div
  className="header"
>
  Music Shop
  <div>
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
     | 
     
    <a
      href="/music"
      onClick={[Function]}
    >
      Music
    </a>
      | 
     
    <a
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
      | 
     
  </div>
</div>
`;
